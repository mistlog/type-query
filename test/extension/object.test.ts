import { getResultCode } from "../common";

describe("object", () => {
    test("pick", () => {
        const name = "object-pick";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

    test("omit", () => {
        const name = "object-omit";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

    test("omit: condition", () => {
        const name = "object-omit-condition";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

    test("readonly", () => {
        const name = "object-readonly";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

    test("optional", () => {
        const name = "object-optional";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

    test("optional: all", () => {
        const name = "object-optional-all";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })
})