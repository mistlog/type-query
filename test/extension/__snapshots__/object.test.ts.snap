// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object omit 1`] = `
"type Todo = {
  title: string;
  description: string;
  completed: boolean;
  \\"temp key\\": string;
};
type TodoPreview = {
  description: string;
  completed: boolean;
};"
`;

exports[`object omit: condition 1`] = `
"type Todo = {
  title: string;
  description: string;
  completed: boolean;
};
type TodoPreview = {
  completed: boolean;
};"
`;

exports[`object optional 1`] = `
"type Todo = {
  title: string;
  description: string;
  completed: boolean;
};
type TodoPreview = {
  title?: string;
  description: string;
  completed: boolean;
};"
`;

exports[`object optional: all 1`] = `
"type Todo = {
  title: string;
  description: string;
  completed: boolean;
};
type TodoPreview = {
  title?: string;
  description?: string;
  completed?: boolean;
};"
`;

exports[`object pick 1`] = `
"type Todo = {
  title: string;
  description: string;
  completed: boolean;
};
type TodoPreview = {
  title: string;
  completed: boolean;
};"
`;

exports[`object readonly 1`] = `
"type Todo = {
  title: string;
  description: string;
  completed: boolean;
};
type TodoPreview = {
  readonly title: string;
  description: string;
  completed: boolean;
};"
`;
